tdp/cli/commands/deploy.py

```diff
@@ -90,14 +90,15 @@ def deploy(
 
         if dry:
             for _ in deployment_iterator:
-                pass
+                _()
             return
 
         # deployment and operations records are mutated by the iterator so we need to
         # commit them before iterating and at each iteration
-        session.commit()  # Update deployment status to RUNNING, operations to PENDING
-        for cluster_status_logs in deployment_iterator:
-            session.commit()  # Update operation status to RUNNING
+        for cluster_status_logs_process in deployment_iterator:
+            session.commit()  # Update deployment and current operation status to RUNNING and next operations to PENDING
+            cluster_status_logs = cluster_status_logs_process()
+            cluster_status_logs
```

Comment:
What is the purpose of this line?

Reply:
Since the __next__ function of the iterator class returns a callable. We have to instatiate each callable in the iterator.